### DISCUSSION TITLE
Linting in VSCode doesn't work, but CLI is fine

### DISCUSSION THREAD

**Discussion by engineertdog** [eslint/eslint#18362] [General]
I've started a new project and swapped from the old usage to this new `eslint/js` library. I have everything working for the most part, and I can lint via the CLI `yarn eslint .` just fine. But I am unable to get linting hints real-time during coding which is somewhat annoying. What piece of information am I missing (the ESLint VSC extension is installed)?

eslint.config.mjs
```js
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";

export default [
  {
    ignores: ["**/*.js"],
  },
  eslint.configs.recommended,
  ...tseslint.configs.recommended,
  {
    rules: {
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": "off",
      "no-undef": "off",
      "@typescript-eslint/no-explicit-any": "off",
      quotes: ["warn", "double", { avoidEscape: true }],
      "eol-last": ["error", "always"],
    },
  },
];
```

On a separate, possibly related note, the ESLint Output tab in VSC states the following:

```
Failed to load the ESLint library for the document c:\path\eslint.config.js
To use ESLint for single JavaScript file install eslint globally using 'npm install -g eslint'.
If you are using yarn or pnpm instead of npm set the setting `eslint.packageManager` to either `yarn` or `pnpm`
You need to reopen VS Code after installing eslint.
```

**Comment by aladdin-add** [comment]
vscode-eslint added flat config supports in `v3.0.5`, but it's a pre-release now: https://github.com/microsoft/vscode-eslint/issues/1644

you can install the version, or using `eslint.experimental.useFlatConfig: true` as of now.

**Reply by dimaslanjaka** [reply]
still not validating after update to latest version

<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/e491e0ee-d9bc-4c06-b444-923432f112a4" />
